src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapper.java

```diff
@@ -122,6 +122,26 @@ public int getNumber() throws AbortException {
     }
 
     @Whitelisted
+    public @Nonnull String getCurrentResult() throws AbortException {
+        Result result = build().getResult();
+        return result != null ? result.toString() : Result.SUCCESS.toString();
+    }
+
+    @Whitelisted
+    public boolean resultIsBetterOrEqualTo(String other) throws AbortException {
+        Result result = build().getResult() != null ? build().getResult() : Result.SUCCESS;
```

Comment:
This would be a race condition were it not for the fact that the `result` can go from null to non-null but not vice-versa.